### DISCUSSION TITLE
Dumping Micro Machines (NES)

### DISCUSSION THREAD

**Discussion by karudo19** [sanni/cartreader#669] [Need help]
Micro Machines is a game by Codemasters. Other Sanni Cart Reader owners have also reported problems dumping games by Codemasters, but I haven't found a similar case to mine.

Mapper (71) is supported, and the game plays great on real hardware with no issues at all. I have firmware version of 7.3 for Sanni Cart Reader v3. All other NES games have been dumped successfully. Whenever I try to dump Micro Machines, I receive a *.bin file with just ê's in it.

I have noticed that only with this cartridge, Cart Reader may not always fully boot into the main menu. But when it does, I can run dumping process completely. Like I have mentioned, no booting issues appear on NES.

Has anyone dumped Micro Machines for NES successfully? Do I need to just change some option(s), update the firmware for Sanni Cart Reader v3, or could it be that this game will become dumpable in the future, so I just have to wait?

Thanks.

**Comment by Cephiros** [comment]
I have never successfully dumped micro machines, but i have been close. I dont remember for sure but i know it dumped completely but a few mytes were different from a good copy. The resulted rom plays but with minor graphics artifacts. I had assumed it was just my copy, i dont know why it would consistantly dump correctly minus a few bytes. If it matters, my micromachines has the gold fingers on back, not the switch.

**Reply by karudo19** [reply]
Thank you for sharing some interesting information. By the way, I forgot to mention that my cartridge is the regular black one.

**Reply by Cephiros** [reply]
Black? Ive only ever seen gold ones. Interesting

**Reply by PsyK0p4T** [reply]
@karudo19 can you : 
1) confirm that your cart is looking like this one : https://nescartdb.com/profile/view/351/micro-machines
2) update your OSCR to latest firmware (v11.5 as we speak)
3) modify manually the settings applied if the game is auto-detected. Set PRG = 256, CHR = 0, RAM = 8, and then try to redump it.
4) let us know what ROM you get then (file size, CRC32, content in hex editor etc.). Eventually compare it to the "known good rom" to find differences

**Reply by karudo19** [reply]
Thank you for your guide. I'll write another reply after I have updated OSCR to the latest firmware and tried again.

**Reply by nsx0r** [reply]
I guess you have a (european?) version that uses the CME-01 chip, which is some sort of copy protection for Codemasters games:
![image](https://user-images.githubusercontent.com/13054953/210127421-ff0472ba-6bdd-4a2d-b2c3-eea5b05c79d4.png)
Dumping these carts with the standard script will only return "0xEA 0xEA 0xEA 0xEA..." because it needs a specific hardware timing that the OSCR doesn't have (yet).

**Comment by PW5190** [comment]
Here's another question. Can the OSCR dump Aladdin Deck Enhancer games, including the original pack-in game _Dizzy the Adventurer_? The Deck Enhancer is an adapter that plays games on special compact cartridges. Only six other games made it into compact cartridge form, and they were all previously released as standalone Codemasters cartridges. I think the way configured is to plug a compact cartridge into the Deck Enhancer, connect the two to the NES adapter, then plug the whole thing into the into the OSCR's SNES cart slot.

**Reply by Cephiros** [reply]
Right, dont know that one, not in my collection unfortunately.

**Reply by nsx0r** [reply]
[yes](https://github.com/sanni/cartreader/discussions/458#discussioncomment-5394388)
